psc-flink/src/main/java/com/pinterest/flink/streaming/connectors/psc/internals/PscConsumerThread.java

```diff
@@ -294,6 +300,10 @@ public void run() {
                     handover.produce(records);
                     records = null;
                 } catch (Handover.WakeupException e) {
+                    Queue<PscEvent> events = handover.getEventQueue();
+                    while (!events.isEmpty()) {
+                        consumer.onEvent(events.poll());
+                    }
```

Comment:
Can you please add a comment for why this block was added?

Reply:
Done, described in PscEvent